### PR TITLE
build: vms and os400 support

### PR DESCRIPTION
**Pull request by jlesk**

### CODE REVIEW

File: http_parser.c [hunk 1]
```diff
@@ -406,6 +397,118 @@ enum http_host_state
...
+    [HTTP_UNSUBSCRIBE] = "UNSUBSCRIBE",
+    [HTTP_PATCH] = "PATCH",
+    [HTTP_PURGE] = "PURGE",
+    [HTTP_MKCALENDAR] = "MKCALENDAR",
+    [HTTP_LINK] = "LINK",
+    [HTTP_UNLINK] = "UNLINK"
+};
+
+
+static const char *errno_name[] = {
```
Comment (indutny) [inline http_parser.c]: These symbols are globally available everywhere, where `http_parser.h` is included. I'm not sure it is a good idea to declare it with such unprefixed names.

**Comment by indutny** [comment]
I'm sorry, but from what it looks like - the code just bloated significantly. I doesn't look like we will accept this patch at the time.

cc @nodejs/http just in case.

**Comment by jlesk** [comment]
Thanks for your comment. The nested macro doesn't work on os400.

**Comment by indutny** [comment]
Is there any chance to use a newer compiler on os400?

**Comment by jlesk** [comment]
I think it is already the most recent one.

**Comment by indutny** [comment]
cc @bnoordhuis maybe

File: http_parser.c [hunk 2]
```diff
@@ -461,15 +564,6 @@ do {                                                                 \
 
 
 /* Map errno values to strings for human-readable output */
```
Comment (craig65535) [inline http_parser.c]: You may want to move this comment

File: http_parser.h [hunk 3]
```diff
@@ -161,58 +150,42 @@ enum flags
  *
  * The provided argument should be a macro that takes 2 arguments.
```
Comment (craig65535) [inline http_parser.h]: Seems this comment is no longer valid

**Comment by bnoordhuis** [comment]
Closing.  Conflicts and unlikely to get merged.  Thanks anyway.

FWIW, http_parser is known to work on z/os nowadays.

**Pull request closed without merging**